### PR TITLE
fix CRC when rssi is enabled

### PR DESCRIPTION
**Pull request by Slubby**

### CODE REVIEW

File: libs/controllers/network/__init__.py [hunk 1]
```diff
@@ -26,24 +28,32 @@ def __init__(self, type: int, message: bytes, source_address: int, destination_a
...
+        if cfg.rssi_enabled:
+            rssi = frame[-1]
+            frame = frame[:-1]
+        else:
+            rssi = -1
+
+        decode_frame = CRC().decode(frame)
+
+        if decode_frame in None:
+            return None
```
Comment (JonaVDM) [inline libs/controllers/network/__init__.py]: Shouldn't this be `is`

File: libs/controllers/network/__init__.py [hunk 2]
```diff
@@ -26,24 +28,34 @@ def __init__(self, type: int, message: bytes, source_address: int, destination_a
...
     def serialize(self) -> bytes:
-        return b''.join([
+        frame = b''.join([
             self.type.to_bytes(1, 'big'),
             self.source_address.to_bytes(2, 'big'),
             self.destination_address.to_bytes(2, 'big'),
             self.data
         ])
 
+        return CRC().encode(frame)
```
Comment (MichelGerding) [inline libs/controllers/network/__init__.py]: do we need to make a new instance every time we encode a message. 
if not would it be possible to move this to the INetworkController.send_message function

File: libs/controllers/network/error/CRC.py [hunk 3]
```diff
@@ -26,7 +26,9 @@ def checksum(self, data: bytes) -> bytes:
         return crc.to_bytes(2, 'big')
 
     def verify(self, data: bytes, checksum: bytes) -> bool:
-        return self.checksum(data) == checksum
+        current_checksum = checksum(data)
+
+        return current_checksum == checksum
```
Comment (MichelGerding) [inline libs/controllers/network/error/CRC.py]: wouldnt it have been easier to just do `return checksum(data) == checksum`

File: libs/controllers/network/__init__.py [hunk 4]
```diff
@@ -89,11 +93,10 @@ def stop(self):
         """ stop the network controller """
         self.task.cancel()
         self.killed = True
-        
 
     def send_message(self, type: int, message: bytes, addr=0xffff):
         """ send a message to the specified address """
-        self.queue.append((type, message, addr))
+        self.queue.append((type, self.crc.encode(message), addr))
```
Comment (JonaVDM) [inline libs/controllers/network/__init__.py]: So were only encoding the body/message part of the frame?

**Review by JonaVDM — APPROVED** [review]

**Review by JonaVDM — APPROVED** [review]

**Pull request merged**